synapseclient/models/mixins/json_schema.py

```diff
@@ -0,0 +1,471 @@
+                semantic_version=json_schema_version.get("semanticVersion", ""),
+                json_sha256_hex=json_schema_version.get("jsonSHA256Hex", ""),
+                created_on=json_schema_version.get("createdOn", ""),
+                created_by=json_schema_version.get("createdBy", ""),
+            ),
+            object_id=response.get("objectId", ""),
+            object_type=response.get("objectType", ""),
+            created_on=response.get("createdOn", ""),
+            created_by=response.get("createdBy", ""),
+            enable_derived_annotations=response.get("enableDerivedAnnotations", False),
+        )
+
+    async def get_schema_async(
+        self, *, synapse_client: Optional["Synapse"] = None
+    ) -> JSONSchemaBinding:
+        """
+        Get the JSON schema bound to the entity.
+
+        Args:
+            synapse_client (Optional[Synapse], optional): The Synapse client instance. If not provided,
+                the last created instance from the Synapse class constructor will be used.
+
+        Returns:
+            JSONSchemaBinding: An object containing details about the bound JSON schema.
+        """
+        response = await get_json_schema_from_entity(
+            synapse_id=self.id, synapse_client=synapse_client
+        )
+        json_schema_version_info = response.get("jsonSchemaVersionInfo", {})
+        return JSONSchemaBinding(
+            json_schema_version_info=JSONSchemaVersionInfo(
+                organization_id=json_schema_version_info.get("organizationId", ""),
+                organization_name=json_schema_version_info.get("organizationName", ""),
+                schema_id=json_schema_version_info.get("schemaId", ""),
+                id=json_schema_version_info.get("$id", ""),
+                schema_name=json_schema_version_info.get("schemaName", ""),
+                version_id=json_schema_version_info.get("versionId", ""),
+                semantic_version=json_schema_version_info.get("semanticVersion", ""),
+                json_sha256_hex=json_schema_version_info.get("jsonSHA256Hex", ""),
+                created_on=json_schema_version_info.get("createdOn", ""),
+                created_by=json_schema_version_info.get("createdBy", ""),
+            ),
+            object_id=response.get("objectId", ""),
+            object_type=response.get("objectType", ""),
+            created_on=response.get("createdOn", ""),
+            created_by=response.get("createdBy", ""),
+            enable_derived_annotations=response.get("enableDerivedAnnotations", False),
+        )
+
+    async def delete_schema_async(
+        self, *, synapse_client: Optional["Synapse"] = None
+    ) -> None:
+        """
+        Delete the JSON schema bound to the entity.
+
+        Args:
+            synapse_client (Optional[Synapse], optional): The Synapse client instance. If not provided,
+                the last created instance from the Synapse class constructor will be used.
```

Comment:
nit: `Arguments:` is preferred over `Args:` (Both should be ok though as far as mkdocs is concerned).

Also, there is no need to add the typing to the docstring - It's implied from the type hint